doc/development/twister.md

```diff
@@ -1,59 +1,281 @@
+- `-y, --dry-run` - Create the filtered list of test cases, but don't actually run them. Useful if
+  you're just interested in the test plan generated for every run and saved in the specified output
+  directory (`testplan.json`).
+- `--overflow-as-errors` - Treat FLASH/RAM/SRAM overflows as errors. **This one is important. If not
+  used the test that overflows is considered to be skipped and not that it has failed.**
+- `-v, --verbose` - Emit build logs, call multiple times to increase verbosity.
+
+## `twister-out` folder
+
+With every invocation of the `east twister` a `twister-out` folder is created. It contains all build
+folders of the created projects, as well as the report files.
+
+Keep in mind, if you keep running `east twister` command and not deleting the `twister-out` folder,
+then the command will rename it to `twister-out.X` (Where `X` is a unique ascending number, starting
+with 1) before running the full build.
+
+## Cookbook style examples
 
-Due to the number of features that it has it can take some time to figure out how to do something
-that you want, see `east twister --extra-help` for the supported options.
+### Running tests on the Host PC
 
-Also, see [Zephyr's documentation](https://docs.zephyrproject.org/latest/develop/test/twister.html)
-for more info.
+You want to build and run all tests under `tests/` directory on the Host PC.
+
+`testcase.yaml` file in the all projects under `tests/` directory should look like this:
+
+```yaml
+tests:
+  tests.test_name.native:
+    platform_allow: native_sim
+    harness: ztest
+```
 
-## Simple example
+**Remember**: Test Scenarios must be unique across the whole repository.
 
 From the project's root directory run:
 
 ```shell
 east twister -p native_sim -T tests
 ```
 
-Twister will detect all projects with `testcase.yaml` in their project root under `tests` directory
-and execute them.
+### Running tests on the target
+
+You want to build and run all tests under `tests/` directory on the target
+
+`testcase.yaml` file in the all projects under `tests/` directory should look like this:
+
+```yaml
+tests:
+  tests.test_name.target:
+    platform_allow: nrf52dk/nrf52832
+    harness: ztest
+```
+
+From the project's root directory run:
+
+```shell
+east twister -T tests -p nrf52dk/nrf52832 --device-testing --device-serial /dev/ttyACM0
+```
+
+### Filtering by tags
+
+You want to only build all projects in the repository that have a tag `some_tag`.
 
-To run the same thing on the target you need to provide more info to the Twister:
+`testcase.yaml` file should then look like this:
+
+```yaml
+tests:
+  samples.bitshifter.with_tag:
+    platform_allow: nrf52dk/nrf52832
+    tag: some_tag
+  samples.bitshifter.without_tag:
+    platform_allow: nrf52dk/nrf52832
+```
+
+From the project's root directory run:
 
 ```shell
-east twister -T tests -p nrf52dk_nrf52832 --device-testing --device-serial /dev/ttyACM0
+east twister -T . -t some_tag
```

Comment:
```suggestion
east twister -T . -t some_tag  --build-only
```

Since you say "only build", did you actually want this?

Reply:
Ah, idea was that this sample demonstrates only a single thing (filtering by tag), but now I see that won't work due to the missing platform flag. I expanded this:

`````
You want to only build (and not run) all projects in the repository that have a tag `some_tag` on
`nrf52dk/nrf52832` target.

`testcase.yaml` file should then look like this:

```yaml
tests:
  samples.bitshifter.with_tag:
    platform_allow: nrf52dk/nrf52832
    tag: some_tag
  samples.bitshifter.without_tag:
    platform_allow: nrf52dk/nrf52832
```

From the project's root directory run:

```shell
east twister -T . -t some_tag -p nrf52dk/nrf52832 --build-only
```
`````